From 1dd44f8e3cbab4913314ee777433825ee9d61867 Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Mon, 4 Feb 2013 12:05:11 +0530
Subject: Both sound and time are abnormal when playback LongDuration_MIDI_5m23s file

BZ: 79222

The mDuration field in MidiFile is not updated until
getDuration is called for the first time. With this change
we are updating mDuration field during prepareAsync call.

Category: AOSP Improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: Yes

Change-Id: Ia95ff143213fb145728b8155858145d63d5f1d0c
Orig-Change-Id: I75aaf92efd41dbcfaf008cc71561ac7b0eda201b
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Signed-off-by: sreekanX <sreekanthx.p.c@intel.com>
Reviewed-on: http://android.intel.com:8080/94263
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Tested-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libmediaplayerservice/MidiFile.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/MidiFile.cpp b/media/libmediaplayerservice/MidiFile.cpp
index 0a6aa90..d827d2d 100644
--- a/media/libmediaplayerservice/MidiFile.cpp
+++ b/media/libmediaplayerservice/MidiFile.cpp
@@ -199,6 +199,13 @@ status_t MidiFile::prepareAsync()
     } else {
         sendEvent(MEDIA_ERROR, MEDIA_ERROR_UNKNOWN, ret);
     }
+    int duration;
+    // mDuration is obtained before the start() call, so that when
+    // playback is started without calling getDuration, it remains updated.
+    // mDuration is updated when getDuration() is called for the first time.
+    if (getDuration(&duration) == NO_ERROR) {
+       ALOGV("Updated mDuration inside getDuration method");
+    }
     return ret;
 }
 
-- 
1.7.4.1

